### PR TITLE
refactor: major clean up of all engines

### PR DESCRIPTION
**Pull request by alan-agius4**
- add return type to domContentLoadedFactory
- remove unused DI tokens
- Remove FileLoader implementations from all engines. Instead use the one in @nguniversal/common
- Extend RenderOptions Express and Hapi engines interfaces from Common Engine RenderOptions

### CODE REVIEW

File: modules/hapi-engine/README.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ You can access them by `@Inject`
 
 ```ts
 import { Request } from 'hapi';
-import { REQUEST } from '@nguniversal/hapi-engine/tokens';
+import { REQUEST } from '@nguniversal/hapi-engine';
```
Comment (CaerusKaru) [inline modules/hapi-engine/README.md:90]: Revert?

File: modules/common/tokens/index.ts [hunk 2]
```diff
@@ -1,8 +0,0 @@
-/**
```
Comment (CaerusKaru) [inline modules/common/tokens/index.ts:1]: Is this technically a breaking change?

Comment (alan-agius4) [reply modules/common/tokens/index.ts:1]: They where marked with `ɵ`. Which indicated that these are private

File: modules/hapi-engine/README.md (hunk 1 above)
Comment (alan-agius4) [reply modules/hapi-engine/README.md]: Done

**Review by CaerusKaru — APPROVED** [review]
Let's battle test this after we merge it to make sure nothing burned down, but LGTM. Nice work!

**Pull request merged**